### DISCUSSION TITLE
Where to download the EV charger usage history?

### DISCUSSION THREAD

**Discussion by avadiax** [netzero-labs/netzero#15] [Q&A] [answered]
Hi @zigam,

I wonder if you can advise if the vehicle charging history (with the breakdown of solar, grid, battery) is done through the FleetAPI or through the Owner API? I have looked through the specs and I wondered where to extract these information?

To further elaborate ... the API "/api/1/energy_sites/{site_id}/calendar_history" downloads the "solar_power", "grid_power" and "battery_power" which then can be used to compute the "load_power". However, I could not find within the APIs where the EV charger power can be downloaded. I see that you have incorporated Vehicle charging history into NetZero and wondered where the information is obtained from.

Appreciate your guidance in this matter.

**Comment by zigam** [comment, marked as answer]
Hi, the data comes from undocumented APIs, so it required some reverse engineering.  At some point I might clean up the code and post it to https://github.com/netzero-labs/tesla-solar-download

**Reply by avadiax** [reply]
The reverse engineering was unexpected ... I thought there are APIs only available for approved developers.

That being the case, I look forward to the update to the code in the future. Thanks for your response.
